src/librustc_resolve/late/diagnostics.rs

```diff
@@ -683,7 +683,16 @@ impl<'a> LateResolutionVisitor<'a, '_, '_> {
                 if let Some(span) = self.r.opt_span(def_id) {
                     err.span_label(span, &format!("`{}` defined here", path_str));
                 }
-                err.span_label(span, format!("did you mean `{}( /* fields */ )`?", path_str));
+                let fields = match self.r.field_names.get(&def_id) {
```

Comment:
```suggestion
                let fields = self.r.field_names.get(&def_id).map_or("/* fields */", |fields| vec!["_", fields.len()].join(", "));
```

Reply:
I had to do `.map_or("/* fields */".to_string(), ..` because `join` gives back a `String`, but otherwise applied.